should now only read in boxy data file

kuziekj opened the pull request:
So this is the start of us addressing the comments in the other PR: https://github.com/mne-tools/mne-testing-data/pull/67

To make things more simple, this PR changes read_raw_boxy so that it now only reads in the raw boxy file, and doesn't require .elp, .mtg, or .evt files. It's also designed to read in either a single boxy file (saved by default as a .txt I believe) or it can read in multiple files (similar to what we had before where it will read the data files for various blocks and montages). Added a new argument to help with this (multi_file=False).

I'll run some tests on this to makes sure style and everything is okay. Feel free to take a look and let me know if anything needs to be changed.

Also, do you think we should add a test file for this to compare the imported data to p_pod? Thought adding that test to this PR would make sense, rather than making a separate PR. Then we could add tests for each new function we add (such as reading .elp, .mtg, or .evt files).

kuziekj committed:
should now only read in boxy data file

kylemath commented:
tutorial crashed, so I commented everything after the loading, and it seems to work, 

I am a bit confused about your question about the test, I think the answer is yes, we want to start with a simple PR with just this and a test of it against Ppod, and then make new PR's to add in all the other features we (you) have made,

kylemath commented:
so maybe we want to brach off importBOXY, save all the stuff you did, and then merge this with importBOXY for this PR

kylemath commented:
also I may have screwed up the commit history above trying to rebase with their master, help please

kylemath committed:
Merge branch 'master' into import_data_only

kylemath committed:
Merge branch 'importBOXY' into import_data_only

kuziekj commented:
Yeah sorry for the confusing question. I meant to ask should this PR only be concerned with loading the raw boxy data, and should a test for that (compare loading to p_pod) also be included. Then should the other functions be their own PR (load .elp, load .evt, etc) and also include tests for those functions. Even though I didn't ask it properly, you answered what I meant to ask.

Yeah I didn't try it with the tutorial beyond the loading. I'm not surprised it crashes since the raw data object is now very bare bones (only containing the data basically). We probably don't want to include the plot_80 tutorial at this point anyway, until we have all the other functions implemented as well.

I'm working on designing a test for the imported data, and working on a tutorial for it as well (similar to plot_30). I'll upload those files later today. I'll also remove the plot_80 tutorial for now.

Okay, I'll try and figure out the rebasing issue. Seems that rebasing can cause lots of problems...

kylemath commented:
Sorry I didn't rebase I was just trying to merge master

On Fri., Jul. 24, 2020, 9:41 a.m. kuziekj, <notifications@github.com> wrote:

> Yeah sorry for the confusing question. I meant to ask should this PR only
> be concerned with loading the raw boxy data, and should a test for that
> (compare loading to p_pod) also be included. Then should the other
> functions be their own PR (load .elp, load .evt, etc) and also include
> tests for those functions. Even though I didn't ask it properly, you
> answered what I meant to ask.
>
> Yeah I didn't try it with the tutorial beyond the loading. I'm not
> surprised it crashes since the raw data object is now very bare bones (only
> containing the data basically). We probably don't want to include the
> plot_80 tutorial at this point anyway, until we have all the other
> functions implemented as well.
>
> I'm working on designing a test for the imported data, and working on a
> tutorial for it as well (similar to plot_30). I'll upload those files later
> today. I'll also remove the plot_80 tutorial for now.
>
> Okay, I'll try and figure out the rebasing issue. Seems that rebasing can
> cause lots of problems...
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/kylemath/mne-python/pull/26#issuecomment-663601258>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA36GFJVRKHV6QUXNWRY6H3R5GTR7ANCNFSM4PFAY6QA>
> .
>

kuziekj commented:
Ah okay. Maybe when we decide to merge, we do a 'squash and merge'?

![image](https://user-images.githubusercontent.com/20359571/88411181-b55f5880-cd94-11ea-802a-7a28c207eeab.png)

kylemath commented:
Yeah


On Fri., Jul. 24, 2020, 10:02 a.m. kuziekj, <notifications@github.com>
wrote:

> Ah okay. Maybe when we decide to merge, we do a 'squash and merge'?
>
> [image: image]
> <https://user-images.githubusercontent.com/20359571/88411181-b55f5880-cd94-11ea-802a-7a28c207eeab.png>
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/kylemath/mne-python/pull/26#issuecomment-663610823>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA36GFKIV64LPEZUTEY3TRDR5GWBLANCNFSM4PFAY6QA>
> .
>

kuziekj committed:
added tutorial and test for loading boxy data, remove plot_80 tutorial and reference for now

kuziekj commented:
Alright so I updated this. Added a tutorial about loading boxy data (similar to plot_30). I added an explanation about how to load and name the montage and block files, if used, but maybe this is a bit too much. Let me know. 

I also added a test file to compare loading boxy data to p_pod.

I removed the plot_80 tutorial for now, and some references to it, since we can't really use it until we have all the other functions implemented. Maybe we also want to change the tutorial since we know more about the ANC dataset.

kylemath commented on doc/overview/datasets_index.rst:
```diff
@@ -230,10 +230,6 @@ Each montage and block contains two marker types:
 - 1 = checkerboard reversal
 - 2 = same as 1 but for the first few trials (to keep separate if needed)
 
```
I think this is from merging to master and we should adjust our PR so this isn't one of the files changed

kuziekj replied:
I did have a reference to the plot_80 tutorial here that I removed.

kylemath commented on tutorials/io/plot_40_reading_boxy_data.py:
```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+r"""
```
why is this r here? why is everything commented below?

kylemath replied:
i see you just copied the test_30 one,
this seems strange

so, I am confused why there are still multiple blocks read in, but maybe it can do both single files and multiple blocks so you kept both those functionalities, it may be cleaner to just make it load single files and run loops over multiple files outside of io/boxy.py and append the Raw objects,

was there a reason you decided to keep the multiple files

also I don't think we should ask people to rename their files here, or at least we should indicate what the files names are like for the raw files saved by boxy, we want to make this tutorial as agnostic to the gratton fabiani techniques as we can

kuziekj replied:
Not really sure what the 'r' is for or why everything is commented, but the other tutorials are like that too (plot_10, 20, 30). Probably something to do with how the html doc is made.

It will read either single .txt files, or multiple files like it did before, based on what the 'multi_file' argument is (True of False). I kept this is because I thought it would be easier to create a single raw data object rather than having to create multiple raw data objects and merge them. I was mostly worried that merging several raw data objects would cause issues, but since they only contain the raw data, maybe that wouldn't be an issue. But yes, just having it always load a single file would be much more straight forward.....I'll change it to keep things simple.

kylemath reviewed (commented):
the changes to boxy.py look good, lets discuss including the multiple files in here,

also if there are events in the column of the data file, does this still parse them and create events? 
I think it should , but maybe we don't have files like that?

kuziekj committed:
now will only load a single boxy file

kuziekj commented:
Alright, updated this so it will only load the default file (.txt) saved by boxy and the tutorial only refers to this file type.

kylemath reviewed (approved):
looks good to me merge and delete when ready, be sure to fork and save the old importBOXY which will have all the other work that we did in this file
